File: app/src/main/java/com/ed/redditapp/ui/fragments/search/SearchPresenter.java
package com.ed.redditapp.ui.fragments.search;

import android.view.View;

public class SearchPresenter {
    private final SearchModel model;
    private final SearchFragment view;

    public SearchPresenter(SearchModel model, SearchFragment view) {
        this.model = model;
        this.view = view;
    }

    public void onSearchTextChanged(String searchText) {
        model.searchSubreddits(searchText, view::updateSearchResults);
    }

    public void onBtnCloseClick(View v) {
        view.close();
    }
}

File: app/src/main/java/com/ed/redditapp/ui/search/SearchPresenter.kt
package com.ed.redditapp.ui.search

import android.view.View

class SearchPresenter(private val model: SearchModel, private val view: SearchFragment) {
    fun onSearchTextChanged(searchText: String) {
        model.searchSubreddits(searchText, view::updateSearchResults)
    }

    fun onBtnCloseClick(v: View?) {
        view.close()
    }

    fun onItemClick(position: Int) {
        view.gotoSubRedditActivity(model.subreddits[position])
    }
}